### ISSUE TITLE
Tiled allows exponent format numbers in polygon coords, TiledSharp does not

### ISSUE THREAD

**Issue by TheMightyGit** [marshallward/TiledSharp#30]
Yeah, it's an edge case, but I hit it by generating coords outside of Tiled.  Tiled was perfectly happy with them, but TiledSharp errored out. Turned out to be choking on the exponent notation. The fix is to add **NumberStyles.AllowExponent** to the **double.Parse** as below to TmxObjectPoint...

```
    public class TmxObjectPoint
    {
        public double X {get; private set;}
        public double Y {get; private set;}

        public TmxObjectPoint(double x, double y)
        {
            X = x;
            Y = y;
        }

        public TmxObjectPoint(string s)
        {
            var pt = s.Split(',');
            X = double.Parse(pt[0], NumberStyles.AllowDecimalPoint | NumberStyles.AllowLeadingSign | NumberStyles.AllowExponent, CultureInfo.InvariantCulture); // NOTE: allow exponent
            Y = double.Parse(pt[1], NumberStyles.AllowDecimalPoint | NumberStyles.AllowLeadingSign | NumberStyles.AllowExponent, CultureInfo.InvariantCulture); // NOTE: allow exponent
        }
    }
```

**Comment by marshallward** [comment]
Thanks, it looks straightforward.  Do you want to just send that in as a pull request?  Otherwise I can put it in.

**Comment by TheMightyGit** [comment]
Sorry for slow response. Could you put it in? Thanks :)

**Comment by marshallward** [comment]
Could you confirm if replacing everything with `NumberStyles.Float` works for you?  Otherwise, I'm sure there will be yet another floating point case that doesn't work.  Also, those lines are getting a little bloated.

**Comment by TheMightyGit** [comment]
Confirmed. NumberStyles.Float works perfectly.

**Comment by marshallward** [comment]
This should be fixed in b10aeec
